### PR TITLE
[ALLUXIO-1813] Only apply spark profile for client artifacts

### PR DESCRIPTION
**Pull request by aaudiber**
https://alluxio.atlassian.net/browse/ALLUXIO-1813

### CODE REVIEW

File: keyvalue/client/pom.xml [hunk 1]
```diff
@@ -50,6 +50,70 @@
     </dependency>
   </dependencies>
 
+  <profiles>
+        <!-- build profile for Apache Spark -->
```
Comment (apc999) [inline keyvalue/client/pom.xml]: interesting indent?

**Comment by apc999** [comment]
one minor format comment left. Thanks for the fix!

File: keyvalue/client/pom.xml (hunk 1 above)
Comment (aaudiber) [inline keyvalue/client/pom.xml]: oops copy pasta

**Comment by aaudiber** [comment]
Moving to branch-1.0

**Pull request closed without merging**